zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/distribution/DbDistributionState.java

```diff
@@ -109,6 +114,18 @@ public DbDistributionState(
             transactionContext,
             continuationByQueueKey,
             new PersistedCommandDistribution());
+
+    metrics = new DistributionMetrics(zeebeDb.getMeterRegistry());
+  }
+
+  @Override
+  public void onRecovered(final ReadonlyStreamProcessorContext context) {
+    final var counter = new AtomicInteger(0);
+
+    // TODO: do it by partition and for each column families
+    // what about performance here? is it okay to take longer during recovery phase?
+    pendingDistributionColumnFamily.forEach(ignore -> counter.getAndIncrement());
+    getMetrics().reset(counter.get());
```

Comment:
this needs to be shaped out and it might be very performance heavy to iterate over all partitions + distributions

Reply:
You could simplify this by using `pendingDistributionColumnFamily.count()`. However as listed on the javadoc of that method, and as you already identified.

> Count the number of entries in the column family by iterating over all its entries. This is an
expensive operation and should be used with care.

Reply:
Yeah, I am unsure if this is feasible, as we need to do `families x partitions (x queues?)` if we want to have counts per queue 🤔

Reply:
Do you know how we do this for other metrics?

Reply:
Yes, for example `BannedInstanceMetrics`, but its only checking one column family

```  @Override
  public void onRecovered(final ReadonlyStreamProcessorContext context) {
    final var counter = new AtomicInteger(0);
    bannedInstanceColumnFamily.forEach(ignore -> counter.getAndIncrement());
    bannedInstanceMetrics.setBannedInstanceCounter(counter.get());
  }
```

Reply:
@npepinpe do you think it's fine to iterate like this? If we wanna use gauges, I don't have another idea how to do it nicely, apart from storing the statistics/count in another column family. 

One further alternative would be to use two counters (one for the `add` operation and one for the `remove` operation per column family) which might mimic a gauge (server side, we can kinda subtract these values from each other). But counters are rather used to describe the rate of change instead of real counts (e.g. they could be reset at any point). Thus, what we could use them for, would be to check if more elements are added to the queue than removed 🤔 

(_disclaimer: I am just learning about these details, thus maybe I am wrong_)

Reply:
> Do we expect this count would ever be very high (e.g. hundreds of thousands)?

No, in a healthy cluster this shouldn't become very high. In theory it'll only grow big if the other partition(s) have a large processing queue, or if communication between brokers somehow fails for a longer period of time. Both of these things are exceptional cases with a large impact on the cluster. I'd question of the slightly longer recovery matters in those situations.

Reply:
Ok, I will move forward with doing the simple `count` per partition per columnfamily and if it causes issues in the future, we need to come up with a better solution.